print-ospf.c

```diff
@@ -611,6 +654,226 @@ ospf_print_tos_metrics(netdissect_options *ndo,
+	    } else if (subtlv_length == 8) {
+		ND_PRINT("\n\t\t  Index: %u, MT-ID: %u, Algorithm: %u, Flags [%s]",
+			 GET_BE_U_4(tptr+4), mt_id, algo,
+			 bittok2str(ep_range_tlv_prefix_sid_subtlv_flag_values, "none", flags));
+	    } else {
+		ND_PRINT("\n\t\tBogus subTLV length %u", subtlv_length);
+	    }
+	    break;
+
+	default:
+	    if (ndo->ndo_vflag <= 1) {
+		if (!print_unknown_data(ndo, tptr, "\n\t\t", subtlv_length))
+		    return -1;
+	    }
+	}
+
+	/* in OSPF everything has to be 32-bit aligned, including subTLVs */
+	if (subtlv_length % 4) {
+	    subtlv_length += (4 - (subtlv_length % 4));
+	}
+	tptr+=subtlv_length;
+	tlv_length-=subtlv_length;
+    }
+    return 0;
+trunc:
+    return -1;
+}
+
+static int
+ospf_ep_lsa_print(netdissect_options *ndo, const uint8_t *tptr, u_int lsa_length)
+{
+    u_int tlv_type, tlv_length;
+    uint16_t range_size;
+    uint8_t af, prefix_length, route_type, flags;
+
+    while (lsa_length >= 4) {
+
+	ND_TCHECK_4(tptr);
+	tlv_type = GET_BE_U_2(tptr);
+	tlv_length = GET_BE_U_2(tptr+2);
+	tptr+=4;
+	lsa_length-=4;
+
+	/* Infinite loop protection. */
+	if (tlv_type == 0 || tlv_length == 0) {
+	    return -1;
+	}
+
+	ND_PRINT("\n\t      %s TLV (%u), length: %u, value: ",
+		 tok2str(lsa_opaque_ep_tlv_values,"unknown",tlv_type),
+		 tlv_type,
+		 tlv_length);
+
+	switch (tlv_type) {
+	case LS_OPAQUE_EP_EXTD_PREFIX_TLV:
+	    ND_TCHECK_8(tptr);
+	    prefix_length = GET_U_1(tptr+1);
+	    af = GET_U_1(tptr+2);
+	    route_type = GET_U_1(tptr);
+	    flags = GET_U_1(tptr+3);
+
+	    if (af != 0) {
+		ND_PRINT("\n\t\tBogus AF %u", af);
+		return -1;
+	    }
+
+	    if (prefix_length > 32) {
+		ND_PRINT("\n\t\tIPv4 prefix: bad bit length %u", prefix_length);
+		return -1;
+	    }
+
+	    ND_TCHECK_4(tptr+8);
+	    ND_PRINT("\n\t\tIPv4 prefix: %15s/%u, Route Type: %s, Flags [%s]",
+		     ipaddr_string(ndo, tptr+4), prefix_length,
+		     tok2str(lsa_opaque_ep_route_type_values, "Unknown", route_type),
+		     bittok2str(ep_tlv_flag_values, "none", flags));
+
+	    /* subTLVs present ? */
+	    if (tlv_length > 12) {
+		if (ospf_print_ep_lsa_extd_prefix_tlv(ndo, tptr+12, tlv_length-12) == -1) {
+		    return -1;
+		}
+	    }
+	    break;
+
+	case LS_OPAQUE_EP_EXTD_PREFIX_RANGE_TLV:
+	    ND_TCHECK_8(tptr);
+	    prefix_length = GET_U_1(tptr);
+	    af = GET_U_1(tptr+1);
+	    range_size = GET_BE_U_2(tptr+2);
+	    flags = GET_U_1(tptr+4);
+
+	    if (af != 0) {
+		ND_PRINT("\n\t\tBogus AF %u", af);
+		return -1;
+	    }
+
+	    if (prefix_length > 32) {
+		ND_PRINT("\n\t\tIPv4 prefix: bad bit length %u", prefix_length);
+		return -1;
+	    }
+
+	    ND_TCHECK_4(tptr+8);
+	    ND_PRINT("\n\t\tIPv4 prefix: %15s/%u, Range size: %u, Flags [%s]",
+		     ipaddr_string(ndo, tptr+8), prefix_length,
+		     range_size,
+		     bittok2str(ep_tlv_flag_values, "none", flags));
+
+	    /* subTLVs present ? */
+	    if (tlv_length > 12) {
+		if (ospf_print_ep_lsa_extd_prefix_tlv(ndo, tptr+12, tlv_length-12) == -1) {
```

Comment:
These 12 should be 8 - you subtracted 4 above.

Reply:
You are correct - fixed in latest commit.